src/index.js

```diff
@@ -268,9 +324,39 @@ module.exports = class Reader extends Component {
     }
   }
   handleWorkerMessage(e) {
-    const { onScan, legacyMode, delay } = this.props
+    const { onScan, legacyMode, delay, showFeedback } = this.props
+    const { preview } = this.els
+
     const decoded = e.data
-    onScan(decoded || null)
+    if(decoded) {
+
+      // Calculate a bounding rectangle around the detected qr code
+      const org_left = Math.min(decoded.location.topLeftCorner.x, decoded.location.bottomLeftCorner.x)
+      const org_right = Math.max(decoded.location.topRightCorner.x, decoded.location.bottomRightCorner.x)
+      const org_top = Math.min(decoded.location.topLeftCorner.y, decoded.location.topRightCorner.y)
+      const org_bottom = Math.max(decoded.location.bottomLeftCorner.y, decoded.location.bottomRightCorner.y)
+
+      // Transform the video-based coordinates into %-based units suitable
+      // for css on a div
+      const {top, left, width, height} = this.transformRect(org_top, org_left, org_bottom, org_right)
+
+      this.setState({
+        qrFound: showFeedback,
+        top: top,
+        left: left,
+        width: width,
+        height: height,
+      })
+    } else {
+      this.setState({
+        qrFound: false,
+        top: 0,
+        left: 0,
+        width: 0,
+        height: 0,
+      })
+    }
+    onScan(decoded && decoded.data || null)
```

Comment:
I think explicit parens here would be nice:

`onScan(decoded && (decoded.data || null))`